ownCloudAppFramework/SDK Extensions/OCQueryCondition+SearchSegmenter.m

```diff
@@ -0,0 +1,304 @@
+	}
+	else if ([segmentStringLowercase isEqual:@":today"])
+	{
+		return ([OCQueryCondition where:OCItemPropertyNameLastModified isGreaterThan:[NSDate startOfRelativeDay:0]]);
+	}
+	else if ([segmentStringLowercase isEqual:@":week"])
+	{
+		return ([OCQueryCondition where:OCItemPropertyNameLastModified isGreaterThan:[NSDate startOfRelativeWeek:0]]);
+	}
+	else if ([segmentStringLowercase isEqual:@":month"])
+	{
+		return ([OCQueryCondition where:OCItemPropertyNameLastModified isGreaterThan:[NSDate startOfRelativeMonth:0]]);
+	}
+	else if ([segmentStringLowercase isEqual:@":year"])
+	{
+		return ([OCQueryCondition where:OCItemPropertyNameLastModified isGreaterThan:[NSDate startOfRelativeYear:0]]);
+	}
+	else if ([segmentStringLowercase containsString:@":"])
+	{
+		NSArray<NSString *> *parts = [segmentString componentsSeparatedByString:@":"];
+		NSString *modifier;
+
+		if ((modifier = parts.firstObject.lowercaseString) != nil)
+		{
+			NSArray<NSString *> *parameters = [[segmentString substringFromIndex:modifier.length+1] componentsSeparatedByString:@","];
+			NSMutableArray <OCQueryCondition *> *orConditions = [NSMutableArray new];
+
+			for (NSString *parameter in parameters)
+			{
+				if (parameter.length > 0)
+				{
+					OCQueryCondition *condition = nil;
+
+					#define OCLocalizedKeyword(x) OCLocalized(x).lowercaseString
+
+					if ([modifier isEqual:@"type"] || [modifier isEqual:OCLocalizedKeyword(@"type")])
+					{
+						condition = [OCQueryCondition where:OCItemPropertyNameName endsWith:[@"." stringByAppendingString:parameter]];
+					}
+					else if ([modifier isEqual:@"after"] || [modifier isEqual:OCLocalizedKeyword(@"after")])
+					{
+						NSDate *afterDate;
+
+						if ((afterDate = [NSDate dateFromKeywordString:parameter]) != nil)
+						{
+							condition = [OCQueryCondition where:OCItemPropertyNameLastModified isGreaterThan:afterDate];
+						}
+					}
+					else if ([modifier isEqual:@"before"] || [modifier isEqual:OCLocalizedKeyword(@"before")])
+					{
+						NSDate *beforeDate;
+
+						if ((beforeDate = [NSDate dateFromKeywordString:parameter]) != nil)
+						{
+							condition = [OCQueryCondition where:OCItemPropertyNameLastModified isLessThan:beforeDate];
+						}
+					}
+					else if ([modifier isEqual:@"on"] || [modifier isEqual:OCLocalizedKeyword(@"on")])
+					{
+						NSDate *onStartDate = nil, *onEndDate = nil;
+
+						if ((onStartDate = [NSDate dateFromKeywordString:parameter]) != nil)
+						{
+							onStartDate = [onStartDate dateByAddingTimeInterval:-1];
+							onEndDate = [onStartDate dateByAddingTimeInterval:60*60*24+2];
+
+							condition = [OCQueryCondition require:@[
+								[OCQueryCondition where:OCItemPropertyNameLastModified isGreaterThan:onStartDate],
+								[OCQueryCondition where:OCItemPropertyNameLastModified isLessThan:onEndDate]
+							]];
+						}
+					}
+					else if ([modifier isEqual:@"smaller"] || [modifier isEqual:OCLocalizedKeyword(@"smaller")])
```

Comment:
String missing `Localizable.strings` file

Reply:
Fixed with the latest commit.